backend/module/eCampCore/src/EntityService/ActivityCategoryService.php

```diff
@@ -2,27 +2,53 @@
 
 namespace eCamp\Core\EntityService;
 
-use Doctrine\ORM\NonUniqueResultException;
 use Doctrine\ORM\ORMException;
 use eCamp\Core\Entity\ActivityCategory;
-use eCamp\Core\Entity\ActivityType;
+use eCamp\Core\Entity\ActivityCategoryTemplate;
 use eCamp\Core\Entity\Camp;
+use eCamp\Core\Entity\ContentTypeConfigTemplate;
 use eCamp\Core\Hydrator\ActivityCategoryHydrator;
 use eCamp\Lib\Acl\NoAccessException;
-use eCamp\Lib\Entity\BaseEntity;
 use eCamp\Lib\Service\EntityNotFoundException;
-use eCamp\Lib\Service\EntityValidationException;
 use eCamp\Lib\Service\ServiceUtils;
 use Laminas\Authentication\AuthenticationService;
 
 class ActivityCategoryService extends AbstractEntityService {
-    public function __construct(ServiceUtils $serviceUtils, AuthenticationService $authenticationService) {
+    /** @var ContentTypeConfigService */
+    protected $contentTypeConfigService;
+
+    public function __construct(
+        ServiceUtils $serviceUtils,
+        AuthenticationService $authenticationService,
+        ContentTypeConfigService $contentTypeConfigService
+    ) {
         parent::__construct(
             $serviceUtils,
             ActivityCategory::class,
             ActivityCategoryHydrator::class,
             $authenticationService
         );
+
+        $this->contentTypeConfigService = $contentTypeConfigService;
+    }
+
+    public function createFromTemplate(Camp $camp, ActivityCategoryTemplate $template) {
+        /** @var ActivityCategory $activityCategory */
+        $activityCategory = $this->create((object) [
```

Comment:
Warum ist hier ein `object` nötig? In `createEntity` wirds dann wieder zu `array` konvertiert. Ist das bereits der übliche Weg wie wir `create` aufrufen?

Reply:
laminas-api-tools ruft den Service mit einem Std-Objekt auf.
Dort wo wir den Service selber aufrufen, machen wird das daher auch so.